shell/common/rasterizer.cc

```diff
@@ -146,6 +178,12 @@ void Rasterizer::DoDraw(std::unique_ptr<flutter::LayerTree> layer_tree) {
   // for Fuchsia to capture SceneUpdateContext::ExecutePaintTasks.
   timing.Set(FrameTiming::kRasterFinish, fml::TimePoint::Now());
   delegate_.OnFrameRasterized(timing);
+
+  if (task_runner_merger_ && task_runner_merger_->DecrementLease()) {
+    return RasterStatus::kEnqueuePipeline;
```

Comment:
This is not trivial to parse, can you add a comment?

Reply:
Yup. It is a tricky case. Added a comment.